specs/transport/car/carv1/index.md

```diff
@@ -89,7 +90,17 @@ Immediately following the Header block, **one or more** IPLD blocks are concaten
 
 #### Length
 
-Each Section begins with a varint representation of an unsigned integer indicating the number of bytes containing the remainder of the section.
+Each Section begins with a unsigned varint representation of an unsigned integer indicating the number of bytes containing the remainder of the section. The length must be an integer greater than zero. Encountering a zero in the length position of a new _Section_ should be considered an error when reading a CAR. Likewise, a CAR writer should only write non-zero length fields.
+
+A CAR is proplerly terminated when an end-of-stream is encountered when attempting to read a new _Section_.
+
+##### Optional mode: null byte as EOF
+
+By default a CAR reader and writer should not support the null byte (`0x00`, varint value of `0`) at the beginning of a _Section_; each _Section_ should begin with a valid unsigned varint of value greater than zero. However, a system using the CAR format may optionally choose to allow the null byte to signal termination of a CAR, allowing it to optionally be followed by arbitrary bytes that are will not considered part of the CAR.
+
+Code libraries that implement the CAR format should treat this mode as an option, and not the default for either reading or writing of the CAR format.
```

Comment:
I agree that the default for writing should not use padding.

I'm less certain about the default for writing. With your wording, then our carv2 implementation allowing padding by default would be going against the spec. It's only going against a "should" and not a "must", but still.

I think the wording should be that decoders "should" treat null as EOF. The only reason we're not doing it in the carv1 library, as I understand it, is that we don't want to break existing users without a semver bump. And doing a semver bump doesn't feel worth it, given we're already doing carv2.

Reply:
Agreed on the reading default.

Reply:
But this is for the CARv1 doc, when read in isolation I would think this is reasonable. I could add an update in this PR to the CARv2 doc and say explicitly that it wraps a CARv1 but enables the optional null-is-EOF mode. Wouldn't that satisfy?

Reply:
Yup that works for me thanks @rvagg